src/main/java/org/opensearch/plugin/insights/core/exporter/LocalIndexExporter.java

```diff
@@ -113,4 +119,47 @@ public void close() {
     private String getDateTimeFromFormat() {
         return indexPattern.print(DateTime.now(DateTimeZone.UTC));
     }
+
+    /**
+     * Set local index exporter data retention period
+     *
+     * @param deleteAfter the number of days after which Top N local indices should be deleted
+     */
+    public void setDeleteAfter(final int deleteAfter) {
+        this.deleteAfter = deleteAfter;
+    }
+
+    /**
+     * Delete Top N local indices older than the configured data retention period
+     */
+    public void deleteExpiredIndices(final Map<String, IndexMetadata> indexMetadataMap) {
+        long expirationMillisLong = System.currentTimeMillis() - TimeUnit.DAYS.toMillis(deleteAfter);
+        for (IndexMetadata indexMetadata : indexMetadataMap.values()) {
+            String indexName = indexMetadata.getIndex().getName();
+            if (!matchesPattern(indexName, indexPattern)) {
+                continue;
+            }
+            if (indexMetadata.getCreationDate() <= expirationMillisLong) {
+                // delete this index
+                client.admin().indices().delete(new DeleteIndexRequest(indexMetadata.getIndex().getName()));
```

Comment:
Also on line 144, instead of `indexMetadata.getIndex().getName()`, just use `indexName` ?

Reply:
Good catch